tools/c7n_gcp/c7n_gcp/resources/compute.py

```diff
@@ -59,41 +64,27 @@ def get_tag_value(self, instance):
         return instance.get('labels', {}).get(self.tag_key, False)
 
 
-class InstanceAction(MethodAction):
-
-    def get_resource_params(self, model, resource):
```

Comment:
the question is if there is any case where these are specific to the action, having the existing method/signature allows for that to be cleanly overridden by denoting its resource type specific may not match up to reality across all services. there's quite a bit api variation across different services.

Reply:
But for now only `Instance` had its class..
And with current implementation its not needed at all..

And its still possible to override it..

Reply:
right but we don't want to introduce new differential behaviors into the generic resource type meta model unless they are exposed across types. this is introducing a differential behavior just for Instance, I'd rather we keep it for the action, and then if refactor to a type method  if its known to be universally of value across other resource types.

Reply:
Its not just for instance..
It will simplify all resources.. in similar manner as I added for disk..

Reply:
we need to try some non compute service resources to verify that this is a good interface. disk is part of compute. try big query or a managed database (postgresql), different teams structure apis different, part of custodian is being flexible enough to get the right abstractions over those.

Reply:
I see that actually `get()` is used with different resource than I used it..
So its not usable at all as I wrote it....